onnx/src/androidMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/objectdetection/SSDMobileNetObjectDetectionModel.kt

```diff
@@ -0,0 +1,61 @@
+package org.jetbrains.kotlinx.dl.api.inference.onnx.objectdetection
+
+import android.graphics.Bitmap
+import org.jetbrains.kotlinx.dl.api.inference.InferenceModel
+import org.jetbrains.kotlinx.dl.api.inference.onnx.OnnxInferenceModel
+import org.jetbrains.kotlinx.dl.api.inference.onnx.executionproviders.ExecutionProvider.CPU
+import org.jetbrains.kotlinx.dl.dataset.Coco
+import org.jetbrains.kotlinx.dl.dataset.CocoVersion.V2017
+import org.jetbrains.kotlinx.dl.dataset.preprocessing.*
+import org.jetbrains.kotlinx.dl.dataset.shape.TensorShape
+
+
+private val SSD_MOBILENET_METADATA = SSDModelMetadata(
+    "TFLite_Detection_PostProcess",
+    "TFLite_Detection_PostProcess:1",
+    "TFLite_Detection_PostProcess:2",
+    0, 1
+)
+
+
+public class SSDMobileNetObjectDetectionModel(override val internalModel: OnnxInferenceModel) :
+    SSDObjectDetectionModelBase<Bitmap>(SSD_MOBILENET_METADATA),
+    InferenceModel by internalModel {
+
+    override val classLabels: Map<Int, String> = Coco(V2017, zeroIndexed = true).labels
+
+    private var targetRotation = 0f
+
+    public constructor (modelBytes: ByteArray) : this(OnnxInferenceModel(modelBytes)) {
+        internalModel.initializeWith(CPU())
+        preprocessing = buildPreprocessingPipeline()
+    }
+
+    override lateinit var preprocessing: Operation<Bitmap, Pair<FloatArray, TensorShape>>
+        private set
+
+    public fun setTargetRotation(targetRotation: Float) {
+        if (this.targetRotation == targetRotation) return
+
+        this.targetRotation = targetRotation
+        preprocessing = buildPreprocessingPipeline()
+    }
+
+    private fun buildPreprocessingPipeline(): Operation<Bitmap, Pair<FloatArray, TensorShape>> {
+        return pipeline<Bitmap>()
+            .resize {
+                outputHeight = inputDimensions[0].toInt()
+                outputWidth = inputDimensions[1].toInt()
```

Comment:
But `inputDimensions` also changes when user runs `reshape`. That's why preprocessing in other models have a getter for preprocessing. If we want to save preprocessing to a variable here for performance reasons, we should also rebuild it when `reshape` is called.

Reply:
I removed the Inference model implementation, so now there is no .reshape method. Also, this model has a fixed input size, so this method didn't make sense